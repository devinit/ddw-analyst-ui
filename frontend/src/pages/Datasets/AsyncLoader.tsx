import * as React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { Dimmer, Loader } from 'semantic-ui-react';

const Datasets: any = React.lazy(() => import('./Datasets')); // TODO: correct variable type
export const AsyncDatasets = (props: RouteComponentProps) => (
  <React.Suspense
    fallback={
      <Dimmer active={true} inverted>
        <Loader content="Loading" />
      </Dimmer>
    }
  >
    <Datasets {...props} />
  </React.Suspense>
);
