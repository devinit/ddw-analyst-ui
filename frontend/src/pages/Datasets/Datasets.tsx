import * as React from 'react';
import { Col, Row } from 'react-bootstrap';
import { List, fromJS } from 'immutable';
import { date, lorem } from 'faker';
import { DatasetsTableCard } from '../../components/DatasetsTableCard';
import { DatasetMap } from '../../types/datasets';

export const Datasets: React.SFC = () => {
  const [ loading, setLoading ] = React.useState(true);
  const [ datasets, setDatasets ] = React.useState(List<DatasetMap>());

  React.useEffect(() => {
    const newDatasets: DatasetMap[] = [];
    for (let i = 0; i < 10; i++) {
      newDatasets.push(fromJS({
        id: `${i}`,
        title: lorem.words(),
        publication: lorem.sentence(),
        releasedAt: date.past()
      }));
    }
    setDatasets(List(newDatasets));
    setLoading(false);
  }, []);

  return (
    <Row>
      <Col>
        <DatasetsTableCard loading={ loading } datasets={ datasets }/>
      </Col>
    </Row>
  );
};

export { Datasets as default };
