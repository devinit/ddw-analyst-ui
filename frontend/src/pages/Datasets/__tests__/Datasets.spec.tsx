/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { Datasets } from '../Datasets';

test('renders the default DataSet page correctly', () => {
  const renderer = TestRenderer
    .create(<Datasets/>)
    .toJSON();

  expect(renderer).toMatchSnapshot();
});
