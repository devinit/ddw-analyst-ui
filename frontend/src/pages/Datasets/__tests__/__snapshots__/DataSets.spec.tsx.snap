// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the default DataSet page correctly 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <div
      className="ui active transition visible inverted dimmer"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="ui text loader"
        >
          Loading...
        </div>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="card-header-text card-header-danger card-header"
      >
        <div
          className="card-title h5"
        >
          <input
            className="w-50 form-control"
            data-testid="dataset-table-search"
            placeholder="Search ..."
          />
        </div>
      </div>
      <div
        className="card-body"
      >
        <div
          className="table-responsive"
        >
          <table
            className="table sources-table table-striped table-hover"
          >
            <thead>
              <tr>
                <th>
                  Title
                </th>
                <th>
                  Publication
                </th>
                <th>
                  Release Date
                </th>
                <th>
                  Actions
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
